test/libsolidity/smtCheckerTests/function_selector/homer.sol

```diff
@@ -0,0 +1,46 @@
+pragma experimental SMTChecker;
+
+interface ERC165 {
+    /// @notice Query if a contract implements an interface
+    /// @param interfaceID The interface identifier, as specified in ERC-165
+    /// @dev Interface identification is specified in ERC-165. This function
+    ///  uses less than 30,000 gas.
+    /// @return `true` if the contract implements `interfaceID` and
+    ///  `interfaceID` is not 0xffffffff, `false` otherwise
+    function supportsInterface(bytes4 interfaceID) external view returns (bool);
+}
+
+interface Simpson {
+    function is2D() external returns (bool);
+    function skinColor() external returns (string memory);
+}
+
+interface PeaceMaker {
+    function achieveWorldPeace() external;
+}
+
+contract Homer is ERC165, Simpson {
+    function supportsInterface(bytes4 interfaceID) public pure override returns (bool) {
+        return
+            interfaceID == this.supportsInterface.selector || // ERC165
+            interfaceID == this.is2D.selector ^ this.skinColor.selector; // Simpson
```

Comment:
```suggestion
            interfaceID == type(ERC165).interfaceId ||
            interfaceID == type(Simpson).interfaceId
```
why not this?

Reply:
Just because then there is no `selector` left in the test :) 
I guess it depends on the meaning of the test, in current for it kind of tests the actual implementation of the interfaceId, which might be a good or a bad thing, I am not sure. After your change, the checks become quite trivial, but maybe that's what is desired?

Reply:
Ah ok, let's leave it like this then.